Source/core/Time.cpp

```diff
@@ -639,19 +639,23 @@ namespace Core {
 
     Time::Time(const uint16_t year, const uint8_t month, const uint8_t day, const uint8_t hour, const uint8_t minute, const uint8_t second, const uint16_t millisecond, const bool localTime)
     {
-        _time.wYear = year;
-        _time.wMonth = month;
-        _time.wDay = day;
-        _time.wHour = hour % 24;
-        _time.wMinute = minute % 60;
-        _time.wSecond = second % 60;
-        _time.wMilliseconds = millisecond;
-        _time.wDayOfWeek = static_cast<WORD>(~0);
-
-        if (localTime) {
-            SYSTEMTIME convertedTime;
-            TzSpecificLocalTimeToSystemTime(nullptr, &_time, &convertedTime);
-            _time = convertedTime;
+        if ((year >= 1900) && ((month >= 1) && (month <= 12)) && ((day >= 1 ) && (day <= 31)) &&
```

Comment:
Can we perhaps make this a bit smarter? Either use a system API to check if the date is valid or otherwise create a small algorithm which checks how many days there are in a month (and also take the leap years into account). There is for example already an is_leap_year method in the cpp file.
And not very important as in the Thunder use case older dates are not relevant but is there a technical reason to limit years > 1900? Don't know if you tested but did a quick search and for the years in struct tm I found "tm_year is a signed value, therefore years before 1900 may be represented.". For the Windows presentation I'm pretty sure it is supported

Reply:
@MFransen69, will add algorithm to validate days in a month case. 
24 will be set to 0, hence why added this case.

Reply:
To my opinion It is not about what (at least the lunix implementation) accepts but what seems logical. Times bigger then 23:59:59.999 do not make much sense or am I missing something?

Reply:
Sure, I will make the valuation accordingly, since in the windows we are doing %24, %60, %60, I added upto the max range. Will restrict the limit to 23:59:59.999

Reply:
Validation logic added for Time inputs as an API and called from both Windows and posix version